hudi-common/src/main/java/org/apache/hudi/common/util/InternalSchemaCache.java

```diff
@@ -217,7 +217,11 @@ public static InternalSchema getInternalSchemaByVersionId(long versionId, String
     }
     InternalSchema fileSchema = InternalSchemaUtils.searchSchema(versionId, SerDeHelper.parseSchemas(latestHistorySchema));
     // step3:
-    return fileSchema.isEmptySchema() ? AvroInternalSchemaConverter.convert(HoodieAvroUtils.addMetadataFields(new Schema.Parser().parse(avroSchema))) : fileSchema;
+    return fileSchema.isEmptySchema()
+            ? StringUtils.isNullOrEmpty(avroSchema)
+              ? InternalSchema.getEmptyInternalSchema()
```

Comment:
Is it because the version upgrade or something? Is the null avro schema coming from an old version Hudi table?

Reply:
@danny0405 Yes, Some users find this problem in the upgrade scenario(0.12.3 -> 0.14).